### ISSUE TITLE
Building with latest oneAPI intel compilers COMPILER_ID == IntelLLVM

### ISSUE THREAD

**Issue by zerothi** [Blosc/c-blosc2#533]
**Describe the bug**
The compilation and detection of the intel oneAPI compiler fails.

**To Reproduce**

```
CC=icx cmake ...
```
fails.

The problem arises since the compiler ID is shown to be IntelLLVM which breaks nearly all of the COMPILER_ID checks present in the CMakeLists file.
```
-- Building for compiler ID IntelLLVM
```
It would be great if this could be amended in some way.
All newer Intel compilers will be based on LLVM and the COMPILER ID will be: `IntelLLVM`.

**System information:**
 - Intel 2023.0.1

**Additional context**
I believe the errors this produce
```
/zhome/0e/2/36189/dcc/bash-build/XeonGold6126/intel/c-blosc2-2.8.0/plugins/filters/bytedelta/bytedelta.c:32:54: error: '__builtin_ia32_palignr128' needs target feature ssse3
```
are only due to the cmake files not being able to detect the features allowed by the compiler. So I think it should be solveable in the cmake files.

Let me know if you need anything else.

**Comment by FrancescAlted** [comment]
Hi there.  Sorry for the late reply.  We don't have resources for supporting compilers beyond gcc/clang and MSVC, but we are receptive to accept PRs for new ones.  If interested, please send yours.

**Comment by FrancescAlted** [comment]
I have merged this, and a new release (2.14.0) will be out soon.
